newm-server/src/main/kotlin/io/newm/server/features/cardano/parser/NFTSongParser.kt

```diff
@@ -160,3 +199,20 @@ internal fun String.toResourceUrl(): String =
             this
         }
     }
+
+internal fun buildNftCdnUrl(
+    fingerprint: String,
+    path: String,
+): String {
+    val url = buildNftCdnUrl(fingerprint, path, "")
+    val hmac = Mac.getInstance("HmacSHA256")
```

Comment:
Can this be a single instance that is re-settable instead of creating a new object each time the method is called?

Reply:
If I do that I'll also need to sync it, because multiple threads can be calling at the same time. I think it would defeat the purpose of reusing to improve performance

Reply:
On second thought, I can create a single instance, initialize it and then clone it here every time. That way we avoid thread sync issues.